cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java

```diff
@@ -176,17 +178,47 @@ private void ensureJDBCDriverIsAvailable() throws Exception {
       LOG.debug("Plugin Type: {} and Plugin Name: {}; Driver Class: {} not found. Registering JDBC driver via shim {} ",
                 dbSinkConfig.jdbcPluginType, dbSinkConfig.jdbcPluginName, driverClass.getName(),
                 JDBCDriverShim.class.getName());
-      driverShim = new JDBCDriverShim(driverClass.newInstance());
+      if (driverShim == null) {
+        driverShim = new JDBCDriverShim(driverClass.newInstance());
+      }
       DBUtils.deregisterAllDrivers(driverClass);
       DriverManager.registerDriver(driverShim);
     }
   }
 
+  private boolean tableNameExists() {
+    try {
+      try {
+        ensureJDBCDriverIsAvailable();
+      } catch (Exception e) {
+        Preconditions.checkArgument(false, "Driver must be able to connect");
+      }
+
+      Connection connection;
+      if (dbSinkConfig.user == null) {
+        connection = DriverManager.getConnection(dbSinkConfig.connectionString);
+      } else {
+        connection = DriverManager.getConnection(dbSinkConfig.connectionString,
+                                                 dbSinkConfig.user, dbSinkConfig.password);
+      }
+
+      DatabaseMetaData metadata = connection.getMetaData();
+      ResultSet rs = metadata.getTables(null, null, dbSinkConfig.tableName, null);
+
+      return rs.next();
```

Comment:
You need to close the `ResultSet` and the `Connection`, otherwise there will be leakage of file descriptors. Do something like this:

```java
try (
  Connection connection = createConnection(...);
  ResultSet rs = connection.getMetaData().getTables(.....);
) {
  return rs.next();
}
```